include/tbb/machine/linux_common.h

```diff
@@ -34,15 +34,19 @@
 #define __TBB_USE_FUTEX 1
 #include <limits.h>
 #include <errno.h>
-// Unfortunately, some versions of Linux do not have a header that defines FUTEX_WAIT and FUTEX_WAKE.
+#include <linux/futex.h>
```

Comment:
By historical reason, `machine/linux_common.h` is included for *BSD systems in [`machine.h`](https://github.com/01org/tbb/blob/tbb_2018/include/tbb/tbb_machine.h#L230-L251). These systems may not have the `linux/futex.h` header file (e.g. they may provide `sys/futex.h`). I would recommend to include `linux/futex.h` only when `__linux__` is defined (in other words, to guard it with `if __linux__ ... endif`).

Reply:
I was under impression that futexes are Linux-specific and expected `SYS_futex` to be not defined on other systems. However, it looks like OpenBSD supports them, but it is not covered by the conditions in `tbb_machine.h`.

I can add conditions for Linux and OpenBSD, but I will also have to disable the futex-related functions below, which presumably should not be defined when futexes are not available.